lib/test/mock_loader.js

```diff
@@ -95,11 +97,17 @@ async function resolve(specifier, context, nextResolve) {
   if (isBuiltin(specifier)) {
     mockSpecifier = ensureNodeScheme(specifier);
   } else {
-    // TODO(cjihrig): This try...catch should be replaced by defaultResolve(),
-    // but there are some edge cases that caused the tests to fail on Windows.
+    const format = defaultGetFormatWithoutErrors(
+      pathToFileURL(context.parentURL ?? specifier)
+    );
+
     try {
-      const req = createRequire(context.parentURL);
-      specifier = pathToFileURL(req.resolve(specifier)).href;
+      if (format === 'module') {
+        specifier = defaultResolve(specifier, context).url;
+      } else {
+        const req = createRequire(context.parentURL);
+        specifier = pathToFileURL(req.resolve(specifier)).href;
```

Comment:
As long as we're comfortable depending on internal things, we should skip the `createRequire` call (which is quite heavy) and call directly the CJS resolver.

Reply:
I wanted to figure out how not to call createRequire, but this would be difficult because the existing cjs filenameResolve relies on a thoroughly generated require.

https://github.com/nodejs/node/pull/53846#discussion_r1677317655
If we think of a direction that does not depend on the internal module, I think this problem can also be solved by creating a new cjs filename resolve.